src/femr/labelers/omop_inpatient_admissions.py

```diff
@@ -185,3 +210,46 @@ def get_outcome_times(self, patient: Patient) -> List[datetime.datetime]:
             if e.code in self.outcome_codes:
                 times.append(e.start)
         return times
+
+    def label(self, patient: Patient) -> List[Label]:
+        """
+        Label all inpatient mortality with prediction_time <= death_time <= discharge
+        Exclude admission if time of discharge or death < prediction time
+        """
+        visits: List[Event] = self.get_visit_events(patient)
+        prediction_start_times: List[datetime.datetime] = [
+            self.visit_start_adjust_func(visit.start) for visit in visits
+        ]
+        prediction_end_times: List[datetime.datetime] = [self.visit_end_adjust_func(visit.end) for visit in visits]
+        outcome_times: List[datetime.datetime] = self.get_outcome_times(patient)
+        death_time = min(outcome_times) if outcome_times else None
+        results: List[Label] = []
+
+        # For each visit, check if there is an outcome which occurs within the (start, end) of the visit
+        for prediction_start, prediction_end in zip(prediction_start_times, prediction_end_times):
+            # exclude if prediction time > discharge time
+            if prediction_start >= prediction_end:
```

Comment:
@guolin1 I made a swap here from ">" to ">=" because I think that's the more correct logic here. Can you double check I did this right?

The idea is that if prediction_start == prediction_end, then it's too easy / trivial of a prediction.

Reply:
That makes sense. Looks good to me! Thanks!